java/src/main/java/com/lancedb/lance/Fragment.java

```diff
@@ -33,6 +35,15 @@ public class Fragment {
     this.fragmentId = fragmentId;
   }
 
+  /** Create a fragment from the given data. */
+  public static FragmentMetadata create(String datasetUri, ArrowArrayStream stream,
```

Comment:
How should we let developer to use this interface tho?

Using `ArrowArrayStream` feels less intuitive for Java developers.

Reply:
Good call! The first level citizens of this API are Presto/Spark, probably need to wrap it in some kind of java stream interface. Previous the HDFS stream interface is the default interface. Now Trino is developing a new interface and there is no a new known interface dominating the whole java ecosystem. @beinan what's your opinion?